crates/holochain/src/core/sys_validate.rs

```diff
@@ -360,14 +360,12 @@ pub fn check_update_reference(
 
 /// Validate a chain of headers with an optional starting point.
 pub fn validate_chain<'iter>(
-    mut headers: impl Iterator<Item = &'iter HeaderHashed>,
+    mut headers: impl Iterator<Item = (&'iter HeaderHash, &'iter Header)>,
```

Comment:
Why did you wind up having to remove `HeaderHashed` everywhere? Seems like `SignedHeader` can coexist with `HeaderHashed`?

Reply:
Because you can't have an iterator with references to a type that `SignedHashed` doesn't have.

Reply:
What do you mean by "a type that `SignedHashed` doesn't have"?

Reply:
`SignedHashed` does not have a `HeaderHashed`. It has the `HeaderHash` and the `T` separately. So I can't give a reference to `HeaderHashed`.